licenses/git_utils.py

```diff
@@ -0,0 +1,41 @@
+import git
+
+
+def setup_local_branch(repo: git.Repo, branch_name: str, parent_branch_name: str):
+    """
+    Ensure we have a local branch named 'branch_name'.
+    Pull from upstream ('origin'), or create as a branch from parent_branch_name.
+    If there's an upstream one, pull from it to make sure we're up to date.
+    Check it out.
+    """
+    # is there already a branch?
+    origin = repo.remotes.origin
+    origin.fetch()
+    if not hasattr(repo.branches, branch_name):
+        # Not locally, maybe upstream
+        if hasattr(origin.refs, branch_name):
+            repo.create_head(branch_name, f"origin/{branch_name}")
+        else:
+            # Nope, need to create from scratch
+            print("create from scratch")
+            parent_branch = getattr(origin.refs, parent_branch_name)
+            repo.create_head(branch_name, parent_branch)
+    else:
+        print("local already exists")
+    branch = getattr(repo.heads, branch_name)
+    branch.checkout()
+    # Make sure local branch is up to date if there's an upstream branch
+    if hasattr(origin.refs, branch_name):
+        origin.pull(f"{branch_name}:{branch_name}")
+
+
+def commit_and_push_changes(repo: git.Repo, branch_name: str, commit_msg: str):
+    """Commit new translation changes to current branch, and push upstream"""
+    index = repo.index
+    index.commit(commit_msg)
+    results = repo.remotes.origin.push(f"{branch_name}:{branch_name}")
```

Comment:
In this function should we use `call_command` and run `publish` for the git branch that has been updated?

Reply:
Indeed, we should!

Reply:
Actually I'm thinking we call publish from the management command, and just have this function do git stuff.

Reply:
copy that